primeui-tester/src/test/java/com/axonivy/ivy/supplements/primeui/tester/TestPrimeUi.java

```diff
@@ -1,166 +1,119 @@
-import org.openqa.selenium.WebElement;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-
-import com.axonivy.ivy.supplements.primeui.tester.PrimeUi.Accordion;
-import com.axonivy.ivy.supplements.primeui.tester.PrimeUi.Dialog;
-import com.axonivy.ivy.supplements.primeui.tester.PrimeUi.SelectBooleanCheckbox;
-import com.axonivy.ivy.supplements.primeui.tester.PrimeUi.SelectCheckboxMenu;
-import com.axonivy.ivy.supplements.primeui.tester.PrimeUi.SelectOneMenu;
-import com.axonivy.ivy.supplements.primeui.tester.PrimeUi.SelectOneRadio;
-import com.axonivy.ivy.supplements.primeui.tester.PrimeUi.Table;
+
+import com.axonivy.ivy.supplements.primeui.tester.widget.Accordion;
+import com.axonivy.ivy.supplements.primeui.tester.widget.Dialog;
+import com.axonivy.ivy.supplements.primeui.tester.widget.SelectBooleanCheckbox;
+import com.axonivy.ivy.supplements.primeui.tester.widget.SelectManyCheckbox;
+import com.axonivy.ivy.supplements.primeui.tester.widget.SelectOneMenu;
+import com.axonivy.ivy.supplements.primeui.tester.widget.SelectOneRadio;
+import com.axonivy.ivy.supplements.primeui.tester.widget.Table;
 import com.codeborne.selenide.Configuration;
-import com.codeborne.selenide.Selenide;
-import com.codeborne.selenide.WebDriverRunner;
 
 /**
  * Class to test PrimeUi. Tests on the official Primefaces Showcase.
- * 
- * @author dhu
- * @since Apr 28, 2016
  */
 public class TestPrimeUi
 {
-  private WebDriver driver;
-  private PrimeUi prime;
-
-  @BeforeEach
-  public void setUp()
+  @BeforeAll
+  public static void setUp()
   {
     Configuration.browser = "firefox";
     Configuration.headless = true;
     Configuration.reportsFolder = "target/senenide/reports";
-    Selenide.open();
-    driver = WebDriverRunner.getWebDriver();
-    driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
-    prime = new PrimeUi(driver);
   }
   
   @Test
-  public void testSelectOneMenu() throws Exception
+  public void testSelectOneMenu()
   {
-    driver.get("http://primefaces.org/showcase/ui/input/oneMenu.xhtml");
-
-    SelectOneMenu selectOne = getSelectOneMenu();
+    open("https://primefaces.org/showcase/ui/input/oneMenu.xhtml");
+    SelectOneMenu selectOne = PrimeUi.selectOne(selectMenuForLabel("Basic:"));
     assertThat(selectOne.getSelectedItem()).isEqualTo("Select One");
     String ps4 = "PS4";
     selectOne.selectItemByLabel(ps4);
     assertThat(selectOne.getSelectedItem()).isEqualTo(ps4);
   }
 
-  private SelectOneMenu getSelectOneMenu()
-  {
-    WebElement selectOne = getElementWithBasicLabel();
-    return prime.selectOne(By.id(selectOne.getAttribute("id")));
-  }
-
   @Test
-  public void testSelectCheckBoxMenu_all() throws Exception
+  public void testSelectCheckBoxMenu_all()
   {
-    driver.get("http://primefaces.org/showcase/ui/input/checkboxMenu.xhtml");
-    
-    getCheckboxMenu().selectAllItems();
-    submitAndCheck("Brasilia");
+    open("https://primefaces.org/showcase/ui/input/checkboxMenu.xhtml");
```

Comment:
this class has two highly duplicated parts:
- url to start with in showcase ( may isolate in a navigator class, but at least introduce a common uri)
- firstItem selection (isolate in a nice utility: maybe combine with navigator) ... e.g.      SelectCheckBoxMenu menu = showcase(inputCHeckbox).first().checkbox();
...assertStuff();